### PR TITLE
[[FIX]] Ignore unused arrow-function parameters if unused: vars

### LINKED ISSUE #2345: Unused arrow-function parameters not ignored by "unused": "vars"
Setting `"unused": "vars"` disables all warnings concerning unused normal function parameters. However the new arrow functions don't seem to be supported by this option:

    on().then( thing => console.log( 'done' ) );

    'thing' is defined but never used.

I feel these should behave the same as normal function parameters when it comes to raising warnings.

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
Fixes #2345

### CODE REVIEW

**Comment by rwaldron** [comment]
Makes sense to me, @caitp @jugglinmike ?

File: src/jshint.js [hunk 1]
```diff
@@ -2880,7 +2880,7 @@ var JSHINT = (function() {
 
     if (loneArg && loneArg.identifier === true) {
```
Comment (caitp) [inline src/jshint.js]: Unrelated, but we should really, really make it clear that `loneArg` is a token --- maybe rename it to clarify that

**Comment by caitp** [comment]
yeah, makes sense to me too

**Comment by jugglinmike** [comment]
Agreed. Thanks, Nicolò!

**Pull request merged**